pilot/pkg/serviceregistry/kube/controller.go

```diff
@@ -564,11 +557,60 @@ func (c *Controller) GetProxyServiceInstances(proxy *model.Proxy) ([]*model.Serv
 				out = append(out, getEndpoints(ss.Addresses, proxyIP, c, port, svcPort, svc)...)
 				out = append(out, getEndpoints(ss.NotReadyAddresses, proxyIP, c, port, svcPort, svc)...)
 			}
+			// If the pod has a Readiness probe, the IP will not be added to Addresses, but to NotReadyAddresses.
+			// That means we may fail to add listeners, resulting in readiness continuing to fail.
+			for _, ea := range ss.NotReadyAddresses {
```

Comment:
this is not needed.. @andraxylia 's code went in. See getEndpoints() stuff above that handles this

Reply:
Yes, resolved the conflict - just need to update the metric.